### PR TITLE
test: add QGB queries test

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Closes https://github.com/celestiaorg/celestia-app/issues/1343

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Review by rootulp — COMMENTED** [review]
LGTM with one small test name fix

File: app/test/qgb_rpc_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+			req: func() error {
+				_, err := queryClient.LastUnbondingHeight(
+					context.Background(),
+					&types.QueryLastUnbondingHeightRequest{},
+				)
+				return err
+			},
+		},
+		{
+			name: "attestation request by nonce",
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply app/test/qgb_rpc_test.go:51]: Thanks a lot :+1: :+1:

**Review by rootulp — APPROVED** [review]

File: app/test/qgb_rpc_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestQGBRPCQueries(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping QGB integration test in short mode.")
+	}
+	_, cctx := testnode.DefaultNetwork(t, time.Millisecond)
+	h, err := cctx.WaitForHeightWithTimeout(401, time.Minute)
```
Comment (evan-forbes) [inline app/test/qgb_rpc_test.go:19]: [definitely optional]
if we don't want to wait a minute here, we can change the genesis state to include a significantly shorter data window param

**Review by evan-forbes — APPROVED** [review]

File: app/test/qgb_rpc_test.go (hunk 2 above)
Comment (SweeXordious) [reply app/test/qgb_rpc_test.go:19]: We won't wait a minute. It's just a worst case scenario. Generally, with a block time of 1ms, 401 blocks will be created in less than half a second.
Yes, we can do that, but ,with fast blocktime, i dont see why tinker with the genesis state and do more changes.

Im okey with any tbh, whichever you prefer :cat:

Comment (evan-forbes) [reply app/test/qgb_rpc_test.go:19]: its fine as is, just an optimization :+1:

Comment (evan-forbes) [reply app/test/qgb_rpc_test.go:19]: block times will likely take longer than a millisecond, just fyi, that's kinda deceiving, as its really just the timeout commit

**Pull request merged**